torch/_subclasses/functional_tensor.py

```diff
@@ -218,12 +235,18 @@ def tolist(self) -> Any:
         else:
             return [elem.tolist() for elem in self.elem]
 
-    def to(self, *args, **kwargs):
-        if _detect_functional_mode().export:
-            # If copy is specified as pos arg, it's always the second one.
-            if len([arg for arg in args if isinstance(arg, bool)]) <= 1:
-                return super().to(*args, **{**kwargs, "copy": True})
-        return super().to(*args, **kwargs)
+    to = _conversion_method_template("to")
+    char = _conversion_method_template("char")
+    cpu = _conversion_method_template("cpu")
+    bfloat16 = _conversion_method_template("bfloat16")
+    byte = _conversion_method_template("byte")
+    cuda = _conversion_method_template("cuda")
+    double = _conversion_method_template("double")
+    float = _conversion_method_template("float")
+    bool = _conversion_method_template("bool")
+    half = _conversion_method_template("half")
+    int = _conversion_method_template("int")
+    long = _conversion_method_template("long")
```

Comment:
Given that you have to override each of these manually anyway, why isn't your override here to translate them to the appropriate .to() call? namely `to(dtype=torch.float, copy=True)` or `to(device="cuda", copy=True)` for example.

Reply:
I looked at the doc of float, and it's indeed pretty simple: https://pytorch.org/docs/stable/generated/torch.Tensor.float.html

I didn't know how stable it would be to translate method like `.float()` to `.to()`, but I guess if that's what you prefer to have, it's worth trying.

Reply:
hmm @albanD I think there are two (semi) independent things here, that I'd be interested in your input on:

(1) (prior to this PR) export was unhappy with the maybe-aliasing behavior of `aten.to()`. They stated that they'd prefer be able to export models with `.to()` in them, and assume that it is non-aliasing, while detecting and banning anyone who uses `.to()` + mutation in a way that would cause this assumption to be wrong (the downside is that some programs might not be exportable, but the hypothesis is that this is rare + fixable with a model tweak)

They got this to work with a small change to functionalization: override `y = x.to(...)` to make it unconditionally copy (there's a kwarg). Then, freeze the storage of `x` and `y`, to ensure that the `copy=True` transformation is semantic-preserving (and export/functionalization errors on any programs that are not semantics-preserving.

(2) (this PR): `aten.to()` is not the only "maybe-aliasing" op. In particular, there are a bunch of tensor methods like `.float()`, `.cpu()`, etc that are also maybe-aliasing, that export would also like to be always-copy.

My question was: if all of those tensor methods were implemented by calling `Tensor.to(...)`, then export would get the behavior they want for free. But they are actually implemented by bypassing the dispatcher and calling `to_impl()`, which requires export to add all of this extra handling in `FunctionalTensor` for each case.

Do you know of any good reason we can't just update those tensor methods to go through the dispatcher and call `aten.to`? (My guess was dispatch overhead, although a single extra hop through the dispatcher seems like a small price to pay).

Reply:
> Do you know of any good reason we can't just update those tensor methods to go through the dispatcher and call aten.to? (My guess was dispatch overhead, although a single extra hop through the dispatcher seems like a small price to pay).

Most likely dispatch overhead yes. We have been historically trying to cut down on framework overhead where we can do so easily and safely.  Most users still use eager so I don't think we want to break that too bad.

> (prior to this PR) export was unhappy with the maybe-aliasing behavior of aten.to(). 

That is due to the fact that this is pre-dispatch IR right? If we were to use post-dispatch, all of this would be gone?

> My question was: if all of those tensor methods were implemented by calling Tensor.to(...)

I don't think that will help with .contiguous() and the like?

Reply:
> That is due to the fact that this is pre-dispatch IR right? If we were to use post-dispatch, all of this would be gone?

In post-dispatch, we still have aliased input-output relationship being baked into the graph, and that's not what we want.

> I don't think that will help with .contiguous() and the like?

I think contiguous() is not a concern to export at least.

Reply:
> I think contiguous() is not a concern to export at least.

Why not? It behaves exactly the same as these functions you're fixing here right?

Reply:
Actually, I think you're right. We should fix .contiguous() in export as well and thanks for pointing that out. 

We do want to decouple addressing an actual use case people need in the short term and potential failures like .contiguous(), thus my previous comment.